main.go

```diff
@@ -276,11 +275,15 @@ func main() {
 	}
 
 	supportedResources := []string{config.ResourceNamePodENI, config.ResourceNameIPAddress}
-	resourceManager, err := resource.NewResourceManager(ctx, supportedResources, apiWrapper)
+	resourceManager, err := resource.NewResourceManager(ctx, supportedResources, apiWrapper, ctrl.Log.WithName("managers").WithName("resource"))
 	if err != nil {
 		ctrl.Log.Error(err, "failed to init resources", "resources", supportedResources)
 		os.Exit(1)
 	}
+	// add health check on subpath for resource manager which includes providers as well
+	for key, value := range resourceManager.GetCheckers() {
+		healthHandler.AddControllerHealthChecker(key, value)
```

Comment:
Synced offline- should we pass the healthHandler to each controller to add health checks?

Reply:
Thanks, I have changed to use parameters.